### ISSUE TITLE
WordPress.com profile changes (display name, email, name, about me) not synced to Atomic

### ISSUE THREAD

**Issue by donalirl** [Automattic/wp-calypso#46077]
#### Steps to reproduce

1. Starting at URL: https://wordpress.com/me and https://wordpress.com/me/account on an atomic site
2. Change profile fields like username, email, name, and about me/ bio
3. Notice that the changes do not reflect on the site or in WP Admin.

#### What I expected

I expected the fields I change in my WordPres.com account to reflect on the site.

#### What happened instead

The two separate profile areas (calypso and wp-admin) do not talk to each other at all. 

#### Context / Source

p1601433483033000-slack-happiness-product

Initially reported and discussed here 308-gh-atomic but moving it to the wp-calypso repo. Fixing this would help smooth out a confusing experience of the atomic platform.

**Comment by katiebethbrown** [comment]
Another report. User had multiple sites that all needed manual updating:

24618223-hc

**Comment by ivan-ottinger** [comment]
Another report: 24630431-hc

→ search and replace did the trick.

**Comment by synora** [comment]
Another report at 3380507-zen

Fixed via search/replace.

**Comment by formosattic** [comment]
25087020-hc

**Comment by donalirl** [comment]
Another instance of confusion in #25390978-hc

**Comment by mikeicode** [comment]
25893381-hc

**Comment by RealCarFax** [comment]
Ran into the same issue here: #420258-hc
Resolved with search and replace command.

**Comment by jamiepalatnik** [comment]
I've noticed this on several chats and tickets in the past as well where the username had to be updated manually.

**Comment by Nic-Sevic** [comment]
3597303-zd-woothemes
manually updated

**Comment by swoodipto** [comment]
Another report: 3687247-zen

The profile picture change in Calypso did not update in WP Admin, guided them to manually change it from WP Admin → Users → Profile.

**Comment by JoshuaGoode** [comment]
Adding that Language settings are also not syncing to the connected user profile.  This has its own report at https://github.com/Automattic/wp-calypso/issues/48980

**Comment by happychait** [comment]
27019743-hc

**Comment by JoshuaGoode** [comment]
Encountered on 27504752-hc

**Comment by NoHopeRadio** [comment]
Encountered on hc-17852724

**Comment by sdnunca** [comment]
Besides the profile fields, the selected language is also not synchronized leading to a very confusing user experience.
I'll try to tackle this during the upcoming HW.

**Comment by pauloeaquino** [comment]
28341100-hc

**Comment by simranvijwani** [comment]
Another report in 28473265-hc

**Comment by devNigel** [comment]
Another report in 28903086-hc

Public Display Name update in Account settings was not getting reflected. Got them to update Nickname in WP Admin and change Public Display Name there.

**Comment by ericamQA** [comment]
28872915-hc Unable to update Display Name without also updating via WP-admin

**Comment by vinnykaur** [comment]
hc-29115264 : Profile pic on email posts different from account profile pic

**Comment by nagpai** [comment]
#29044319-hc

Causing confusion to the user. Username changed in calypso, but  People >> team and WP admin (CLI ) show older username.

**Comment by jcortx** [comment]
#29268011-hc

**Comment by poojashetty18** [comment]
Another report: 3976503-zen

**Comment by jamiepalatnik** [comment]
Another report in 21013947-hc

**Comment by donalirl** [comment]
Confusion in #29291820-hc. Updating the WordPress.com profile name in https://wordpress.com/me did not change the author name on the site. It wasn't clear that the name needed to be changed under Users → Advanced Users Management.

**Comment by katiebethbrown** [comment]
Report in 29631185-hc

User updated username and display name on wpcom and wasn't seeing those changes reflect on the live site b/c wp-admin didn't pick them up.

**Comment by allilevine** [comment]
This can be fixed in a similar way to [syncing the user locale](https://github.com/Automattic/jetpack/pull/19525):
- Make the user's profile fields available to Jetpack sites (see D59872-code)
- Sync the user's profile fields in Jetpack
- Clear the user data cache every time a user updates their profile fields (see D61323-code)

**Comment by rickmgithub** [comment]
Another report - #26042544-hc
I changed the display name in cli for them. Now I'm seeing the other options above but user was happy.

**Comment by cpapazoglou** [comment]
Info not synced between Calypso & Atomic:
- Username (can only be changed in Calypso)
- First Name
- Last Name
- Display Name
- Email
- About me

Info currently in sync between Calypso & Simple / Atomic:
- Profile Picture (through Gravatar)
- Interface Language
- Color Scheme

**Comment by cpapazoglou** [comment]
This is a little trickier than anticipated. For example:

- User A changes display name in WordPress.com to **user_A_new_name**
- User B, visits User A posts.
- User B should be seeing **user_A_new_name** as the author of posts.

Since xmlrpc requests (D59872-code) are done for the current user (User B) the display name of User A won't be updated unless User A visits the site as authenticated user.

So, instead of flushing the user's data cache we should exec an async job that updates the user's options.
Also, we should hide the fields in WP Admin and instead show to the user a WordPress.com url to change these details.
![](https://cln.sh/xXBIBE+)

**Comment by cpapazoglou** [comment]
Just stashing some code D61740-code

**Comment by cpapazoglou** [comment]
I have update the code here D61740-code, it's ready for review. 

After a short convo (p1621940313143500/1621935668.137700-slack-C01VDMPHG69) about hiding the profile fields from WP Admin, @mmtr suggested:
- Hide these fields for all WP.COM users editing their own profile
- Hide these fields for administrators trying to edit another WP.COM user
- DO NOT Hide these fields for administrators trying to edit a user not connected to WP.COM

The above changes will be in a separate Jetpack PR.

**Comment by cpapazoglou** [comment]
Since https://github.com/Automattic/wp-calypso/issues/50709, I had some second thoughts on this. Please read more here pd2C0z-1T-p2

**Comment by cpapazoglou** [comment]
Deployed r226375-wpcom for **testing** the async jobs in production

**Comment by cpapazoglou** [comment]
Deployed r226450-wpcom that enables syncing to Atomic.

**Next Step**: Hide these fields from `wp-admin/profile.php` for WordPress.com connected users.

**Comment by cpapazoglou** [comment]
Here is the Simple sites part of hiding the fields D62166-code

**Comment by cpapazoglou** [comment]
And here is the Atomic part https://github.com/Automattic/jetpack/pull/20007

**Comment by cpapazoglou** [comment]
Deployed r226947-wpcom for hiding fields in Simple sites

**Comment by cpapazoglou** [comment]
Merged https://github.com/Automattic/jetpack/pull/20007, profile fields on Atomic WP-Admin will be hidden after the next jetpack release on July 6, 2021.

**Comment by katiebethbrown** [comment]
30446950-hc

Display name not updated.

**Comment by kathrynwp** [comment]
Ran into this again in 4111200-zen

**Comment by BogdanUngureanu** [comment]
~~Hey @katiebethbrown and @kathrynwp, can you confirm that both encounters were on Atomic? :) If that's the case, the changes introduced in the merged PR will be available on Atomic after the next Jetpack release, which is tomorrow.~~

**Comment by BogdanUngureanu** [comment]
Created D63715-code that fixes the issue.

**Comment by BogdanUngureanu** [comment]
The issue should be fixed - deployed a fix in r228231-wpcom. However, there's another issue that strips emojis and diacritics from the profile information. For this, there's an additional patch that's currently pending for review: D63730-code

**Comment by BogdanUngureanu** [comment]
Deployed the emoji and diacritics fix in rWP228360-wpcom

**Comment by kaitohm** [comment]
@BogdanUngureanu Since this has been deployed, we've encountered an issue, which I originally posted here: https://github.com/Automattic/wp-calypso/issues/54403

### Issue

User bio is no longer locatable by a third-party theme, that used to pull in the bio from the wp-admin user info.

### Proposal

Could it be that the location we're syncing and storing the user bio now in wp-admin is different, so themes that rely on that are no longer able to locate the bio info?

I'm not sure if we should continue this conversation here, or if it should be on that other bug report I made, linked above.

**Comment by kaitohm** [comment]
I have another case: 30542562-hc (Follow up: 4128986-zd-woothemes )

### Issue

Display name and Gravatar updated in WordPress.com profile settings is not updating the wp-admin Display Name and Profile Picture. And due to this deploy, users aren't able to edit them in wp-admin either: https://github.com/Automattic/jetpack/pull/20007

### Partial Workaround

An HE can update a user's display name in CLI with `user update user_id --display_name=new_user_name`.

However, the Profile picture cannot be updated.

Related internal discussion: p1625884673152900-slack-C02FMH4G8

**Comment by kaitohm** [comment]
Regarding the two issues mentioned above - the situation seems to have resolved itself now. It could have been site caching involved, too. I can no longer reproduce this issue on my test sites either, so I will close the issue.

**Comment by edwinho89** [comment]
Just experienced this issue in 4134359-zd-woothemes

It appears the issue remains, and current workaround using CLI is the way to go:
`user update user_id --display_name=new_user_name` 

Reopen the GH for visibility.

**Comment by jordesign** [comment]
Another instance of this (Profile image only) in #4135256-zen

**Comment by NoHopeRadio** [comment]
In 4130013-zen, they could not update their author description. Updated it via CLI `user update user_id --description=<description>`

**Comment by sbathompson-he** [comment]
Reported here: 30778529-hc
Used `user update user_id --description="description"` to update it for the user and let them know

**Comment by BogdanUngureanu** [comment]
@jordesign The picture (Gravatar) is managed using Gravatar, not with the implementation in the context of this issue. From what I've noticed, it looks like it's a caching issue with Gravatar. Would it be possible to open up a separate issue for this? 

@edwinho89 I've created a differential that fixes this scenario and I'll deploy it tomorrow on WPCOM, I'll let you know as soon as that's done, thanks for the report. :)

Besides this, I've also opened a PR in Jetpack that should handle a possible scenario that could've happened in the extra reports from this issue. Sadly, I can't guarantee that would fix all issues. As a reference, the sync will still fail in the following scenarios:
- Jetpack connection issues.
- On account email change, when a user already exists with that e-mail address in the Atomic site (but only for the email field will not be updated).

**Comment by mmtr** [comment]
> @edwinho89 I've created a differential that fixes this scenario and I'll deploy it tomorrow on WPCOM, I'll let you know as soon as that's done, thanks for the report. :)

This has been deployed to production today in r229672-wpcom.

> I've also opened a PR in Jetpack that should handle a possible scenario that could've happened in the extra reports from this issue.

Merged as well today, however this won't be shipped to production until next September 7 once Jetpack 10.1 is out.

**Comment by formosattic** [comment]
The merge of profile makes sense if there's only one site, but users with multi-sites might want to have a separate profile (Public display name) for each site.

Even if we were to enable the Display Name fields in WP Admin, it sounds like it'd be challenging (and overkill?) to find a way to reflect that in WordPress.com. 

A quick workaround is to add another admin user by:

* creating another WordPress.com account, then invite it to the site
OR
* creating another user from WP Admin, then edit the field "Display name publicly as"

New user profile can then be selected from the editor (under `Post > Status & visibility > Author`).

User report: 4221284-zd-woothemes

**Comment by jordesign** [comment]
Just re-opening this as the fix is still inbound in the next Jetpack release on Sep6. I'll test and close the issue once the fix is confirmed.

**Comment by retnonindya** [comment]
I'm commenting on this for reference/documentation purposes. I'm not expecting things will suddenly change overnight, but we might be able to consider this in future development.

Just today, @flameslash and I discussed an issue where a user (named X) would like to have a different public display username for their two websites: One simple site, one AT site.

Slack discussion: p1630890874323500-slack-C03TY6J1A

The goal is, for the user (X) to use public display name X on their AT site, and public display name Y on their simple site.

This echoed @formosattic's thought:

> The merge of profile makes sense if there's only one site, but users with multi-sites might want to have a separate profile (Public display name) for each site.

It _used to be_ possible with AT's feature on having its own system of updating public display username. Change user X's public display username to Y from WordPress.com/me, then on the AT site, change the public display username back to X. 

In the end, we suggested the user create a new WordPress.com account (account Y with public display name Y) and add it to the simple site. This is faster and might be safer/reduce confusion.

However, yes, we have discussed changes on the Profile area, and it might be too much to revert things back/overkill. Also, this might be some edge cases (involving WPCOM users only. Local site/WP Admin users should not be affected) so the workaround should still be feasible.
